### DISCUSSION TITLE
Windows Media Player as rending engine

### DISCUSSION THREAD

**Discussion by LvDaniel** [AntonyCorbett/OnlyM#401] [Ideas]
Hi Antony,
Will you consider to add Windows Media Player as rending engine to OnlyM?
I'm asking this because on our KH PC using Media Foundation videos are not running fluently. They seem to stutter and this is more visible in scenes where is more action. I did some tests using Media Foundation with --nogpu but it's getting worst. Also when using ffmpeg as rending engine video it's freezing for 1 second or so from time to time so cannot be an option for us. 
While running using Media Foundation, CPU is around 70% and GPU at 50-60%.
Our KH PC is an Dell Inspiron 3252 with Intel Pentium N3700 processor (low power consumption). This has been acquired through LDC 2 years ago and won't be replaced soon.
In Soundbox we were using Windows Media Player as video player and everything worked smoothly. CPU was less stressed ~40-50%.
Last night at memorial for example we need to play a video so I used SoundBox for playing it and OnlyM for displaying images. Fortunately they are running ok together but will be nice if we could rely just on OnlyM when we need to play extra videos and images. 
I don't know what rending engine JW Library is using but this also works normally.
Thanks for considering this.

**Comment by Blann** [comment]
I’m fairly sure JW Library uses Media Foundation in some form. When I globally disabled MF to check something else, nothing would play from JWL.

However there are definitely some disparities in the rendering between JWL and OnlyM – there’s a faded, washed-out appearance to videos rendered with MF in OnlyM on our PC which is not present in JWL. Mentioned this to Antony a while back but we couldn’t find what was causing it. Given this it does seem entirely possible that there could be performance differences too.

**Comment by rainking430** [comment]
Our KH has noticed video performance issues with OnlyM as well. We use an Intel NUC with a core i3 processor. JW Library, Soundbox, and VLC are all smooth as silk, but OnlyM stutters. VLC would be a nice media rendering option too if that is possible, as it's installed on our PC per the Branch's "Using VLC for Video Playback" document.

**Comment by blazer2004** [comment]
The JW Library-VLC-JWmediawin combo is a dream. Excellent video playback. You don't have to close JW library anymore to playback VLC, true playlisting with mark in and mark out. VLC isn't nearly as good as the pro grade mac solutions that use a QuickTime backend and cost an arm and a leg, but it works good for the price. PowerPoint has a awesome video playback component as well, but there are some videos in the wild it could handle better.

That said, I have helped with a install where VLC didn't second screen correctly, triple- and quad checking the setup instructions, and we had to fall back to OnlyM. The PC that was provided wasn't short on CPU either.

**Comment by rainking430** [comment]
Thanks for the tips. Didn't know about JwlMediaWin, will revisit using VLC with that. I also think there may have been some confusion here as to whether VLC can handle still images, but from what I looked up it seems that it can handle them quite easily.

**Comment by blazer2004** [comment]
VLC handles images just fine

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by LvDaniel** [comment]
@blazer2004 I've tried to use VLC together with JW Library and JwlMediaWin but doesn't work unless I manually minimize JW Library Media Window. Is there another way?
I assume that VLC needs to be configured to be Fullscreen on Display2 and also to use an undocked playlist, isn't it?

**Comment by Blann** [comment]
You need to uncheck the Always on Top option in JwlMediaWin.

**Comment by LvDaniel** [comment]
@blazer2004 thanks for the tip, it works. I see a flash on the 2nd screen when video starts running and when stops (probably due to video being set to full screen) but I assume is normal.

**Comment by AntonyCorbett** [comment]
@LvDaniel Have you tried installing K-Lite codecs?

**Comment by LvDaniel** [comment]
@AntonyCorbett no I haven't. Is there something I need to configure after the install? Thank you.

**Comment by AntonyCorbett** [comment]
Configuration is done during install. I've used it a few times to fix poor video playback on some machines.

**Comment by AntonyCorbett** [comment]
The installer provides system-wide codecs. The uninstaller completely removes them so it's safe to experiment with

**Comment by LvDaniel** [comment]
Thanks for the tip, I'll give it a try. I assume the rending engine in OnlyM will remain Media Foundation, right?

**Comment by AntonyCorbett** [comment]
@LvDaniel yes

**Comment by LvDaniel** [comment]
I've installed K-Lite codecs but unfortunately I haven't seen any improvements.

**Comment by AntonyCorbett** [comment]
@LvDaniel There should be some performance improvement in the MediaFoundation using the latest version of OnlyM (which uses the .NET 5 runtime). I still have the option to implement a VLC renderer.
